tracing/tags.go

```diff
@@ -11,8 +9,19 @@ var alwaysIncludeAllowList = []string{
 	TagKeyEndpoint,
 }
 
-// actual type: []string
-var tagsAllowList atomic.Value
+// NOTE: In general storing a pointer of []string to an atomic value does not
+// really guarantee atomicy because a slice is a container and its content can
+// be modified (e.g. replace individual strings inside). A copy of the slice is
+// required to guarantee atomicy.
+//
+// But here, when storing it we always create a new string (via
+// generateAllowList), and the read function (getAllowList) is unexported and we
+// never mutate the returned result, so skip copying is OK.
+//
+// But if any of those assumptions are broken in the future, we can not storing
+// the atomic.Pointer[[]string] naively, and would need to either make a copy or
```

Comment:
🔕 What you could also do is store an `atomic.Pointer[string]` and do a split whenver you need to access it.  Splitting a string into a slice is quite fast, and in some cases you might even get escape analysis to notice the slice isn't leaked.

(only works if there is a safe way to use some delimiter without messing up the tags)